library/src/androidTest/java/org/xmtp/android/library/GroupMembershipChangeTest.kt

```diff
@@ -0,0 +1,96 @@
+import org.xmtp.android.library.messages.walletAddress
+import uniffi.xmtpv3.org.xmtp.android.library.codecs.GroupMembershipChangeCodec
+import uniffi.xmtpv3.org.xmtp.android.library.codecs.GroupMembershipChanges
+
+@RunWith(AndroidJUnit4::class)
+class GroupMembershipChangeTest {
+    lateinit var fakeApiClient: FakeApiClient
+    lateinit var alixWallet: PrivateKeyBuilder
+    lateinit var boWallet: PrivateKeyBuilder
+    lateinit var alix: PrivateKey
+    lateinit var alixClient: Client
+    lateinit var bo: PrivateKey
+    lateinit var boClient: Client
+    lateinit var caroWallet: PrivateKeyBuilder
+    lateinit var caro: PrivateKey
+    lateinit var caroClient: Client
+    lateinit var fixtures: Fixtures
+
+    @Before
+    fun setUp() {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        fixtures =
+            fixtures(
+                clientOptions = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    enableAlphaMls = true,
+                    appContext = context
+                )
+            )
+        alixWallet = fixtures.aliceAccount
+        alix = fixtures.alice
+        boWallet = fixtures.bobAccount
+        bo = fixtures.bob
+        caroWallet = fixtures.caroAccount
+        caro = fixtures.caro
+
+        fakeApiClient = fixtures.fakeApiClient
+        alixClient = fixtures.aliceClient
+        boClient = fixtures.bobClient
+        caroClient = fixtures.caroClient
+    }
+
+    @Test
+    fun testCanAddMembers() {
+        Client.register(codec = GroupMembershipChangeCodec())
+
+        val group = alixClient.conversations.newGroup(
+            listOf(
+                bo.walletAddress.lowercase(),
+                caro.walletAddress.lowercase()
+            )
+        )
+        val messages = group.messages()
+        assertEquals(messages.size, 1)
+        val content: GroupMembershipChanges? = messages.first().content()
+        assertEquals(
+            listOf(bo.walletAddress.lowercase(), caro.walletAddress.lowercase()).sorted(),
+            content?.membersAddedList?.map { it.accountAddress.lowercase() }?.sorted()
+        )
+        assert(content?.membersRemovedList.isNullOrEmpty())
+    }
+
+    @Test
+    fun testCanRemoveMembers() {
+        Client.register(codec = GroupMembershipChangeCodec())
+
+        val group = alixClient.conversations.newGroup(
+            listOf(
+                bo.walletAddress.lowercase(),
+                caro.walletAddress.lowercase()
+            )
+        )
+        val messages = group.messages()
+        assertEquals(messages.size, 1)
+        assertEquals(group.memberAddresses().size, 3)
+        group.removeMembers(listOf(caro.walletAddress.lowercase()))
+        assertEquals(group.messages().size, 2)
+        assertEquals(group.memberAddresses().size, 2)
+        val content: GroupMembershipChanges? = messages.first().content()
```

Comment:
Shouldn't we be looking for the second item in the array here, not the first?

Reply:
No because the list is reversed because the messages come in descending order.